cal-itp/benefits
# Feat: CalFresh re-enrollment error

Closes #1921.

~Currently on top of #1968 since both of these PRs introduce migrations.~

~Currently on top of #1989 since both of these PRs work on the enrollment views.~

~Currently on top of #2053 since both of these PRs introduce migrations.~

## How to test

1. (Temporarily) override the value returned by [`session.enrollment_expiry()`](https://github.com/cal-itp/benefits/blob/dev/benefits/core/session.py#L110) e.g.

    ```python
    expiry = datetime(2024, 5, 31, tzinfo=timezone.utc).timestamp()
    if expiry:
        return datetime.fromtimestamp(expiry, tz=timezone.utc)
    else:
        return None
    ```

1. In the Admin, update a flow that you can pass eligibility for (e.g. MST Courtesy Cards): check supports expiration and add a value for the expiration days, re-enrollment days, and re-enrollment error template = `enrollment/reenrollment-error--calfresh.html`
1. Pass eligibility
1. Enter the path `/enrollment/reenrollment-error`
1. See the new re-enrollment error page
